plugins/guests/haiku/cap/configure_networks.rb

```diff
@@ -0,0 +1,29 @@
+require "tempfile"
+
+require_relative "../../../../lib/vagrant/util/template_renderer"
+
+module VagrantPlugins
+  module GuestHaiku
+    module Cap
+      class ConfigureNetworks
+        include Vagrant::Util
+
+        def self.configure_networks(machine, networks)
```

Comment:
You'll want to get the list of interfaces here using the :network_interfaces capability, so you can set the device names, something like:

```ruby
interfaces = machine.guest.capability(:network_interfaces)
networks.each do |network|
  network[:device] = interfaces[network[:interface]]
  ...
```
```